File: AutoStock.Domain/Entities/AuditLog.cs
namespace AutoStock.Domain.Entities;

public class AuditLog
{
    public int Id { get; set; }

    public string? UserId { get; set; }

    public string? UserEmail { get; set; }

    public string Action { get; set; }
        = string.Empty;

    public string EntityName { get; set; }
        = string.Empty;

    public string? EntityId { get; set; }

    public string? Details { get; set; }

    public DateTime CreatedAt { get; set; }
        = DateTime.UtcNow;
}

File: AutoStock.Client/src/types/auditLog.ts
export interface AuditLog {
    id: number;
  
    userId: string | null;
  
    userEmail: string | null;
  
    action: string;
  
    entityName: string;
  
    entityId: string | null;
  
    details: string | null;
  
    createdAt: string;
  }
  
  
  export interface AuditLogsResponse {
    items: AuditLog[];
  
    page: number;
  
    pageSize: number;
  
    totalCount: number;
  
    totalPages: number;
  
    hasPreviousPage: boolean;
  
    hasNextPage: boolean;
  }
  
  
  export interface AuditLogQuery {
    search?: string;
  
    action?: string;
  
    entityName?: string;
  
    page?: number;
  
    pageSize?: number;
  }